### PR TITLE
Allow users with no friends to log in successfully

### PR DESCRIPTION
**Pull request by tomalley104**
**NOTE**: Branched off of work done in #8, so review/merge that first so that only relevant changes are seen here. This was done in an attempt to avoid doing double the work when rebasing/merging this later. 

In `FirebaseManager.swift`:
- Primary work was done in `FirebaseManager.getContacts(:)`, that was where the bug was.
- Also ended up refactoring the class a bit— I didn't touch the insides of any methods, but mostly just rearranged the properties, and organized the methods by grouping relevant ones together with pragma marks. 

Moving forward, I think we should touch this class as little as possible- there's a lot that could go wrong with Firebase, (primarily the lack of callback when it cant connect, leaving the method hanging _forever_), so I really think the first priority post-deadline is to figure out a different solution for storing user data. What's nice is that we can store the data basically the same way, but get a bunch of conveniences for how we get it by using a server. (eg - dont have to fetch each username then user, we can just ask the server for all of this user's friends. one callback, _thats it_ ).

### CODE REVIEW

File: Thirty/Thirty/Utilities/FirebaseManager.swift [hunk 1]
```diff
@@ -247,66 +182,69 @@ class FirebaseManager {
...
-                        UserManager.shared.contacts.append(user)
-                        dispatchGroup.leave()
-                    }
-                }
-                dispatchGroup.notify(queue: DispatchQueue.global(qos: .`default`)) {
-                    DispatchQueue.main.async {
-                        UserManager.shared.contacts.sort(by: { $0.username.lowercased() < $1.username.lowercased() })
-                        completion(.Success)
-                    }
+        let cancelBlock: (Error) -> Void = { completion(.failure($0) )}
```
Comment (alanscarpa) [inline Thirty/Thirty/Utilities/FirebaseManager.swift]: the firebase cancelBlock has me a little confused - what's the purpose/when would it get called?  It's not called if there's a firebase error, correct?

File: Thirty/Thirty/Utilities/Result.swift [hunk 2]
```diff
@@ -9,36 +9,36 @@
 import Foundation
 
 public enum Result<T> {
```
Comment (alanscarpa) [inline Thirty/Thirty/Utilities/Result.swift]: taken right from the intrepid playbook

File: Thirty/Thirty/Utilities/FirebaseManager.swift [hunk 3]
```diff
@@ -333,25 +271,117 @@ class FirebaseManager {
...
         }) { (error) in
-            completion(.Failure(error))
+            completion(.failure(error))
         }
     }
+
+    // MARK: Device Token
+
+    func updateDeviceToken() {
+        databaseRef.child("users").child(UserManager.shared.currentUserUsername.lowercased()).updateChildValues(["device-token": TokenUtils.deviceToken])
```
Comment (alanscarpa) [inline Thirty/Thirty/Utilities/FirebaseManager.swift:286]: We use "device-token" string in multiple places.  We might want to make these childValues constants?

Comment (tomalley104) [reply Thirty/Thirty/Utilities/FirebaseManager.swift:286]: yeah there's a lot of low hanging fruit like this in the firebase manager, i think we should deliberate and do all of the models/constants in one sweep. I want to check out leveraging `Codable` for doing a lot of this stuff.

File: Thirty/Thirty/Utilities/FirebaseManager.swift (hunk 1 above)
Comment (tomalley104) [reply Thirty/Thirty/Utilities/FirebaseManager.swift]: It only gets fired when the user doesnt have _permission_ to perform that request/access that data. Basically if firebase deems your action "illegal" and thus cancels your action

**Pull request merged**